great_expectations/experimental/datasources/filesystem_data_asset.py

```diff
@@ -138,72 +52,15 @@ def build_batch_request(
-
-    def get_batch_list_from_batch_request(
-        self, batch_request: BatchRequest
-    ) -> List[Batch]:
-        self._validate_batch_request(batch_request)
-        batch_list: List[Batch] = []
-
-        kwargs: dict | None = getattr(self, "kwargs", None)
-        if not kwargs:
-            kwargs = {}
 
-        for request, path in self._fully_specified_batch_requests_with_path(
-            batch_request
-        ):
-            batch_spec = PathBatchSpec(
-                path=str(path),
-                reader_method=self._get_reader_method(),
-                reader_options=self.dict(
-                    include=self._get_reader_options_include(),
-                    exclude=self._EXCLUDE_FROM_READER_OPTIONS,
-                    exclude_unset=True,
-                    by_alias=True,
-                    **kwargs,
-                ),
-            )
-            execution_engine: PandasExecutionEngine | SparkDFExecutionEngine = (
-                self.datasource.get_execution_engine()
-            )
-            data, markers = execution_engine.get_batch_data_and_markers(
-                batch_spec=batch_spec
-            )
-
-            # batch_definition (along with batch_spec and markers) is only here to satisfy a
-            # legacy constraint when computing usage statistics in a validator. We hope to remove
-            # it in the future.
-            # imports are done inline to prevent a circular dependency with core/batch.py
-            from great_expectations.core import IDDict
-            from great_expectations.core.batch import BatchDefinition
-
-            batch_definition = BatchDefinition(
-                datasource_name=self.datasource.name,
-                data_connector_name="experimental",
-                data_asset_name=self.name,
-                batch_identifiers=IDDict(request.options),
-                batch_spec_passthrough=None,
-            )
-
-            batch_metadata = copy.deepcopy(request.options)
-            batch_metadata["base_directory"] = path
+        return super().build_batch_request(options)
 
-            # Some pydantic annotations are postponed due to circular imports.
-            # Batch.update_forward_refs() will set the annotations before we
-            # instantiate the Batch class since we can import them in this scope.
-            Batch.update_forward_refs()
-            batch_list.append(
-                Batch(
-                    datasource=self.datasource,
-                    data_asset=self,
-                    batch_request=request,
-                    data=data,
-                    metadata=batch_metadata,
-                    legacy_batch_markers=markers,
-                    legacy_batch_spec=batch_spec,
-                    legacy_batch_definition=batch_definition,
-                )
-            )
-        self.sort_batches(batch_list)
-        return batch_list
+    def _get_data_connector(self) -> DataConnector:
```

Comment:
The data connector constructor should be lightweight (see previous comment), and the instance of the DataConnector need could then be initialized in the constructor of FilesystemDataAsset and merely returned here.

Here we are creating a new DataConnector with each call to `_get_data_connector`, which would have performance implications especially given that the data connector manages the cache.